tests/subsys/settings/src/settings_enc.c

```diff
@@ -1,236 +0,0 @@
-/*
```

Comment:
Can we move removal of this file to other PR?
I am not an expert in settings and I do not understand what are these tests for, but maybe instead of throwing them out we should make them actually run?

Reply:
> Can we move removal of this file to other PR?

Sure.

> I am not an expert in settings and I do not understand what are these tests for, but maybe instead of throwing them out we should make them actually run?

From what I can tell, `settings_enc.c` was intended to test `subsys/settings/src/settings_line.c` implementation. This is a middle/helper layer for FCB and File storage backends, where entries are written in a `name=value` manner, called "lines" in its implementation. Unfortunately both FCB and File backends differ in behavior of how entries (lines) are written to backend, i.e. in case of File backend each "line" is prepended with 2-byte length,  while in case of FCB there is nothing more prepended. This different behavior for FCB vs File is selected at build time by `bool SETTINGS_ENCODE_LEN` Kconfig option. This means that checking output of such "line encoder" is not possible in a shared module under `tests/subsys/settings/src/` directory, but should rather be done for each backend separately or a new directory/sample should be created to test both behaviors.

Additionally `settings_enc.c` seems to be expecting base64 encoded values, which are no longer produced and were removed as part of 55be783c85f4 ("settings: Remove deprecated SETTINGS_USE_BASE64 support").

So overall it seems more feasible to write tests for `settings_line.c` module from scratch or at most be inspired by current implementation. (assuming that current FCB and File specific tests, which use this functionality, are not enough).

And as written in the commit message, this `settings_enc.c` was *never* used/tested.